### PR TITLE
Translate apart of  floatingpoint.po

### PR DESCRIPTION
**Pull request by yichung279**

### CODE REVIEW

**Comment by adrianliaw** [comment]
Hi @yichung279, although I haven't review the pull request yet (thank you so much for keep contributing!! ✨😄), I've noticed that you've kept using the same branch for the previous PR, please do *NOT* reuse a previous branch unless you've recreated it.  Please *always* follow the steps described in the **每一次翻譯時** section in https://github.com/python/python-docs-zh-tw#id3, where you should fetch from upstream (do *NOT* use git pull) and create a branch from `upstream/3.7`.

For now it's ok and I'll selectively merge your commits, no worries! Just remember to do so in your next PR 😉

I'll review it in a bit.

File: tutorial/floatingpoint.po [hunk 1]
```diff
@@ -105,6 +105,10 @@ msgid ""
 "fraction is ``3602879701896397 / 2 ** 55`` which is close to but not exactly "
 "equal to the true value of 1/10."
 msgstr ""
+"只要你停在任何有限的數或位數，您就只會得到近似值。而現在大多數的計算機中，浮"
```
Comment (adrianliaw) [inline tutorial/floatingpoint.po]: 這邊我覺得只需要說「有限的位數」就可以了。

File: tutorial/floatingpoint.po [hunk 2]
```diff
@@ -105,6 +105,10 @@ msgid ""
...
+"點數是透過二進位分數近似的，其中分子從最高有效位元使開始用53個位元表示，分母"
```
Comment (adrianliaw) [inline tutorial/floatingpoint.po]: 53 旁邊要有空格

File: tutorial/floatingpoint.po [hunk 3]
```diff
@@ -114,18 +118,30 @@ msgid ""
 "if Python were to print the true decimal value of the binary approximation "
 "stored for 0.1, it would have to display ::"
 msgstr ""
+"由於數值顯示的方式，很多使用者並沒有發現數值的近似。Python 只會印出一個十進位"
```
Comment (adrianliaw) [inline tutorial/floatingpoint.po]: 我覺得「數值的近似」也許如果翻成「數值是個近似值」會比較順一點，原因是「數值的近似」這樣的文法讓「近似」可以是形容詞轉名詞，或是動詞轉名詞；如果以動詞轉名詞的方式解釋就沒有問題，但如果以形容詞轉名詞解釋的話，這句話的意思就會像是 "Many users are not aware of the similarity"。我有點這樣的感覺

File: tutorial/floatingpoint.po [hunk 4]
```diff
@@ -114,18 +118,30 @@ msgid ""
...
+"\n"
+"::"
 
 #: ../../tutorial/floatingpoint.rst:71
 msgid ""
 "That is more digits than most people find useful, so Python keeps the number "
 "of digits manageable by displaying a rounded value instead ::"
 msgstr ""
+"這比一般人感到有用的位數還多，所以 Python 將位數保持在可以接受的範圍，只顯示"
+"簡化後的數值：\n"
```
Comment (adrianliaw) [inline tutorial/floatingpoint.po]: 「簡化」可以再更貼近原文一點，原文是用 "rounded value" 也就是四捨五入。

File: tutorial/floatingpoint.po [hunk 5]
```diff
@@ -114,18 +118,30 @@ msgid ""
...
+"\n"
+"::"
 
 #: ../../tutorial/floatingpoint.rst:77
 msgid ""
 "Just remember, even though the printed result looks like the exact value of "
 "1/10, the actual stored value is the nearest representable binary fraction."
 msgstr ""
+"一定要記住，雖然印出的數字看起來是精準的 1/10，但真正儲存的數值是能表示的二進"
+"位分數中，最接近的精準數值的數。"
```
Comment (adrianliaw) [inline tutorial/floatingpoint.po]: 「最接近的精準數值的數」中第一個「的」是不是能省略掉？

File: tutorial/floatingpoint.po [hunk 6]
```diff
@@ -137,6 +153,11 @@ msgid ""
 "values share the same approximation, any one of them could be displayed "
 "while still preserving the invariant ``eval(repr(x)) == x``."
 msgstr ""
+"有趣的是，有許多不同的十進位數，共用同一個最接近的二進位近似小數。例如說：數"
+"字 ``0.1`` 和 ``0.10000000000000001`` 和 "
+"``0.1000000000000000055511151231257827021181583404541015625``，都由 "
+"``3602879701896397 / 2 ** 55`` 近似。由於這三個數值共用同一個近似值，任何一個"
+"數值都可以被顯示，同時保持不變量 ``eval(repr(x)) == x``。"
```
Comment (adrianliaw) [inline tutorial/floatingpoint.po]: 我覺得 "invariant" 在這邊好像沒有特別加上什麼意思，我自己是覺得如果把「不變量」拿掉可能甚至會比較好理解。

File: tutorial/floatingpoint.po [hunk 7]
```diff
@@ -145,6 +166,9 @@ msgid ""
 "Starting with Python 3.1, Python (on most systems) is now able to choose the "
 "shortest of these and simply display ``0.1``."
 msgstr ""
+"歷史上，Python 的提示字元 (prompt) 與內建的 :func:`repr` 函式會選擇上段說明中"
+"有17個有效位元的數：``0.10000000000000001``。從 Python 3.1 版開始，Python（在"
```
Comment (adrianliaw) [inline tutorial/floatingpoint.po]: 17 旁邊要有空格

File: tutorial/floatingpoint.po [hunk 8]
```diff
@@ -154,12 +178,18 @@ msgid ""
 "arithmetic (although some languages may not *display* the difference by "
 "default, or in all output modes)."
 msgstr ""
+"注意，這是二進位浮點數理所當然的特性，並不是 Python 的錯誤 (bug)，更不是您程"
+"式碼的錯誤。只要有程式語言支持硬體的浮點數運算，您將會看到同樣的事情出現在其"
+"中（雖然某些程式語言預設不 *顯示* 差異，或者預設全部輸出）。"
```
Comment (adrianliaw) [inline tutorial/floatingpoint.po]:
```suggestion
"中（雖然某些程式語言預設不\\ *顯示*\\ 差異，或者預設全部輸出）。"
```

這邊需要加上這個倒斜線來拿掉空格，不過雖然這邊再 po 檔裡這樣寫，在 poedit 之類的軟體中打開的話只會看到一個斜線

File: tutorial/floatingpoint.po [hunk 9]
```diff
@@ -154,12 +178,18 @@ msgid ""
...
 
 #: ../../tutorial/floatingpoint.rst:99
 msgid ""
 "For more pleasant output, you may wish to use string formatting to produce a "
 "limited number of significant digits::"
 msgstr ""
+"為求更優雅的輸出，您可能想要使用字串的格式化產生限定的有效位數：\n"
```
Comment (adrianliaw) [inline tutorial/floatingpoint.po]: 這邊可以幫我在「字串的格式化」後面加上括弧來註明原文是 string formatting 嗎？

**Review by adrianliaw — COMMENTED** [review]
Hi @yichung279 Sorry for the procrastination, I really like your translations and enjoyed reviewing it very much! There are a few details need to be tweaked, please take a look at the code review when you have a minute. Thank you so much for translating this, this is a rather difficult passage to translate!

**Comment by yichung279** [comment]
OK, I will fix them asap.

File: tutorial/floatingpoint.po (hunk 4 above)
Comment (yichung279) [reply tutorial/floatingpoint.po]: 那我根據[教育研究院的辭典](http://terms.naer.edu.tw/search/?q=rounding&field=ti&op=AND&group=&num=10)，翻譯為捨入。

File: tutorial/floatingpoint.po (hunk 6 above)
Comment (yichung279) [reply tutorial/floatingpoint.po]: 好，通順為主。

File: tutorial/floatingpoint.po (hunk 9 above)
Comment (adrianliaw) [reply tutorial/floatingpoint.po]: @yichung279 這邊可以再幫忙補上嗎？感謝你！

**Comment by adrianliaw** [comment]
@yichung279 Thank you! I've manually taken out merge commits and then squash-and-merged the pull request, now closing this.

**Pull request closed without merging**